### PR TITLE
solve both the problems

### PR DESCRIPTION
**Pull request by sinscary**
This program is written in Ruby

To run please cd into root folder of project and run

`ruby main.rb`

This will run the program and write the output in csv files inside fixtures folder.

##### Note - This will run solution for both the problems. Also you can update the csv files as per your test cases.%

### CODE REVIEW

**Comment by sinscary** [comment]
@maknahar Can you review my solution, please?

File: services/calculation_service.rb [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    @input_data = input_data
+    @result = []
+  end
+
+  def calculate_minimal_cost
+    @input_data.each do|row|
+      delivery_possible = false
+      mincost = ""
+      delivery_partner = ""
+      @partners.each do|partner|
```
Comment (svetha-cvl) [inline services/calculation_service.rb]: Try simplifying this by using ruby's enumerable and map/reduce methods like map and reject

Comment (sinscary) [reply services/calculation_service.rb]: @svetha-cvl made the changes. Can you please review it again?

**Comment by sinscary** [comment]
@maknahar Can you review the changes please?

**Pull request closed without merging**